mmpose/models/heads/heatmap_heads/heatmap_head.py

```diff
@@ -64,6 +64,8 @@ class HeatmapHead(BaseHead):
             keypoint coordinates from the network output. Defaults to ``None``
         init_cfg (Config, optional): Config to control the initialization. See
             :attr:`default_init_cfg` for default settings
+        extra (dict, optional): Extra configurations.
+            Defaults to ``None``
```

Comment:
The argument `extra` is convenient for extending the head class but may confuse users. We should keep a well-defined interface where every argument has a clear meaning and a detailed usage description in the docstring.

Here I think maybe we can use existing arguments `conv_out_channels`, `conv_kernel_sizes`, and `has_final_layer` to configure the final conv layers.